shards/shards.go

```diff
@@ -270,6 +270,12 @@ func (tl *loader) load(keys ...string) {
 	// finished running shardedSearcher will be ready.
 	defer tl.ss.markReady()
 
+	if len(keys) == 0 {
+		// If there's nothing to load, we exit early here, but we want to mark
+		// ourselves as ready.
+		return
+	}
```

Comment:
this is unrelated to the PR and breaks a fix that has landed. Similarly with the change to watcher.go

Reply:
I removed those changes - I had brought them in with a rebase or merge from main - and I rebased with changes that have been merged with main since this branch was created. Of note, I merged [#516](https://github.com/sourcegraph/zoekt/pull/516) - there were some conflicts to resolve.